src/gradescopeapi/_config/config.py

```diff
@@ -2,63 +2,54 @@
 Configuration file for FastAPI. Specifies the specific objects and data models used in our api
 """
 
-import requests
-from enum import Enum
+from datetime import datetime
+from typing import Optional
 from pydantic import BaseModel
 
 
-class SemesterModel(str, Enum):
-    spring = "spring"
-    summer = "summer"
-    fall = "fall"
-    winter = "winter"
+class LoginRequestModel(BaseModel):
+    email: str
+    password: str
 
 
-class SessionModel(BaseModel):
-    session: requests.Session
+class CourseID(BaseModel):
+    course_id: str
 
 
-class ConnectionModel(BaseModel):
-    session: SessionModel
-    logged_in: bool = False
+class AssignmentID(BaseModel):
+    course_id: str
+    assignment_id: str
 
 
-class TokenModel(BaseModel):
-    access_token: str
+class AssignmentDates(BaseModel):
+    course_id: str
+    assignment_id: str
+    release_date: Optional[datetime] = None
+    due_date: Optional[datetime] = None
+    late_due_date: Optional[datetime] = None
 
 
-class CourseModel(BaseModel):
+class StudentSubmission(BaseModel):
+    student_email: str
     course_id: str
-    course_short_name: str
-    course_name: str
-    course_semester: SemesterModel
-    course_year: int
-    num_grades_published: int
+    assignment_id: str
 
 
-class UserModel(BaseModel):
-    user_id: str
-    user_name: str
-    user_email: str
+class ExtensionData(BaseModel):
+    course_id: str
+    assignment_id: str
 
 
-class AssignmentStatusModel(str, Enum):
-    submitted = "submitted"
-    not_submitted = "not submitted"
-    graded = "graded"
+class UpdateExtensionData(BaseModel):
+    course_id: str
+    assignment_id: str
+    user_id: str
+    release_date: Optional[datetime] = None
+    due_date: Optional[datetime] = None
+    late_due_date: Optional[datetime] = None
```

Comment:
Should stick to a consistent style

```suggestion
    release_date: datetime | None = None
    due_date: datetime | None = None
    late_due_date: datetime | None = None
```